source/base/parameter_handler.cc

```diff
@@ -647,6 +647,35 @@ namespace Patterns
   }
 
 
+  void Selection::string_to_any(const std::string &s, boost::any &v) const
+  {
+    AssertThrow(match(s),
+                ParameterHandler::ExcValueDoesNotMatchPattern(s,description()));
+    if (v.type() == typeid(std::string *))
+      *(boost::any_cast<std::string *>(v)) = s;
+    else
+      {
+        AssertThrow(false, ExcIncompatibleType(v, *this));
+      }
+  }
+
+
+  std::string
+  Selection::any_to_string(const boost::any &v) const
+  {
+    std::string s;
+    if (v.type() == typeid(const std::string *))
+      s = *(boost::any_cast<const std::string *>(v));
+    else
+      {
+        AssertThrow(false, ExcIncompatibleType(v, *this));
+      }
+    AssertThrow(match(s),
+                ParameterHandler::ExcValueDoesNotMatchPattern(s,description()));
```

Comment:
And the same here. This will throw if the string stored in `v` does not match the selection.